packages/forms/src/model.ts

```diff
@@ -926,7 +925,7 @@ export abstract class AbstractControl {
    */
   private _parentMarkedDirty(onlySelf?: boolean): boolean {
     const parentDirty = this._parent && this._parent.dirty;
-    return !onlySelf && parentDirty && !this._parent._anyControlsDirty();
+    return !onlySelf && !!parentDirty && !this._parent!._anyControlsDirty();
```

Comment:
Just a suggestion... wouldn't it be better to do it like below and removing the line above it in order to avoid an unnecessary variable, any coercion and the non-null assertion operator?

```suggestion
    return this._parent && !onlySelf && this._parent.dirty && !this._parent._anyControlsDirty();
```

Reply:
Yeah, I noticed that it's awkwardly written as is, but wanted to avoid introducing any refactoring and concentrate only on the the bare minimum, especially since it's a breaking change.